relations/src/proc_macro/README.md

```diff
@@ -0,0 +1,113 @@
+# liminal-ark-relation-macro
+
+This crate provides `snark_relation` procedural macro for concise defining SNARK relations.
+Given minimal relation definition, this macro will generate all the required code for creating and casting partial relation objects, public input serialization and circuit generation.
+
+## General usage
+
+The `#[snark_relation]` attribute is intended for modules. Such module must define two items:
+  1. *relation object*: the collection of all constant, public and private relation data. The
+     struct should be defined with `#[relation_object_definition]` attribute.
```

Comment:
```suggestion
     struct must be defined with `#[relation_object_definition]` attribute.
```
?
Since there's no other way it would work. _Should_ sounds as if it was optional.

Reply:
done